### PR TITLE
gentoo: swith to llvm (clang)

### PR DESCRIPTION
**Pull request by 257**
This will pave the way for actual arch support and cross-compiling

### CODE REVIEW

File: mkosi/distributions/gentoo.py [hunk 1]
```diff
@@ -147,6 +148,11 @@ def install(cls, state: MkosiState) -> None:
                     "--exclude", "./dev",
                     "--exclude", "./proc",
                 ])
+            # gnuefi on clang will trigger this: https://bugs.gentoo.org/887047
+            # REMOVE when https://github.com/systemd/systemd/pull/26641 is released
+            with (config / "package.accept_keywords/9999").open("a") as f:
+                f.write("=sys-apps/systemd-9999 **")
```
Comment (behrmann) [inline mkosi/distributions/gentoo.py:154]: Probably missing a newline?

**Comment by 257** [comment]
forgot to comment out `Repository=` in ci to actually test the builds with clang (binaries under my repo are built with clang though).

**Review by behrmann — COMMENTED** [review]
How common is it to build Gentoo with LLVM instead of GCC and why can't GCC not do cross-compilation?

**Comment by 257** [comment]
> How common is it to build Gentoo with LLVM instead of GCC

no idea. there is no stats on that.

> and why can't GCC not do cross-compilation?

gcc:
https://wiki.gentoo.org/wiki/Cross_build_environment
https://wiki.gentoo.org/wiki/Embedded_Handbook/General/Creating_a_cross-compiler

clang, none of the above.

**Comment by DaanDeMeyer** [comment]
I think you're going to have to update the binpkgs repo first, this is way too slow for CI now

**Comment by 257** [comment]
out dated gcc, updating...

**Comment by 257** [comment]
> How common is it to build Gentoo with LLVM instead of GCC and why can't GCC not do cross-compilation?

no so bad read on this: [Cross compiling made easy, using Clang and LLVM](https://mcilloni.ovh/2021/02/09/cxx-cross-clang/)

**Comment by DaanDeMeyer** [comment]
Can you paste the error from source build run into somewhere regarding /sys? I'd prefer to fix the cause instead of merging workarounds if possible

**Comment by 257** [comment]
> Can you paste the error from source build run into somewhere regarding /sys? I'd prefer to fix the cause instead of merging workarounds if possible

https://github.com/systemd/mkosi/pull/1665

**Comment by 257** [comment]
> Can you get me the contents of config.log? That should tell us what specifically is failing. Also, is this with or without LLVM?


off your `gentoo-bwrap`: https://gist.github.com/257/6a174cd8fe2d29ab78d085963e2741d0

**Comment by DaanDeMeyer** [comment]
Let's close this until glibc supports compilation with LLVM

**Comment by DaanDeMeyer** [comment]
I can't reopen this one, can you recreate?

**Pull request closed without merging**